feat: add custom callback example and reference colab

parambharat opened the pull request:
## Description

This PR adds details to the huggingface documentation related to custom logging and callbacks. Additionally, we also fix the description of the `WANDB_LOG_MODEL` env var to reflect the new changes in the HF integration.

## Checklist

Check if your PR fulfills the following requirements. Put an `X` in the boxes that apply.

- [X] Files I edited were previewed on my local development server with `yarn start`. My changes did not break the local preview.
- [X] Build (`yarn docusaurus build`) was run locally and successfully without errors or warnings.
- [X] I merged the latest changes from `main` into my feature branch before submitting this PR.

parambharat committed:
feat: add custom callback example and reference colab

morganmcg1 commented on docs/guides/integrations/huggingface.md:
```diff
@@ -346,6 +346,58 @@ wandb.init(project="amazon_sentiment_analysis",
 ### Custom logging
 
 Logging to Weights & Biases via the [Transformers `Trainer` ](https://huggingface.co/transformers/main\_classes/trainer.html)is taken care of by the `WandbCallback` ([reference documentation](https://huggingface.co/transformers/main\_classes/callback.html#transformers.integrations.WandbCallback)) in the Transformers library. If you need to customize your Hugging Face logging you can modify this c ...
+For instance here's how you could modify the callback to log predictions to a Weights & Biases table after each evaluation
```
1. Can you add a header to this section: “View evaluation samples during training”
2. Can you complete the code example to show the callback being added to the Trainer
3. Show how to set an optional logging frequency if they don’t want to log every eval_steps
4. A bit more explanation of what is going on in this code

parambharat replied:
@morganmcg1 , Thanks for the detailed review comments. I have addressed all of them. Please let me know what you think and resolve the conversation if you are satisfied.

parambharat committed:
fix: add details for the following

1. add a header to this section: “View evaluation samples during training”
2. complete the code example to show the callback being added to the Trainer
3. Show how to set an optional logging frequency if they don’t want to log every eval_steps
3. A bit more explanation of what is going on in this code

parambharat committed:
fix: typo in using the tokenizer

parambharat committed:
fix: add missing frequency in the class attributes

morganmcg1 reviewed (approved):
LGTM, thanks for the changes!

morganmcg1 commented:
@ngrayluna would love a merge on this whenever you get a chance, as lukas has been asking for it

morganmcg1 committed:
Merge branch 'main' into hf/custom-callback

morganmcg1 commented:
fixed merge conflicts

ngrayluna reviewed (changes requested):
Build failing. Markdown format needs fix.

Commit:
updates eval callback example, refreshes docs, adds table image
